src/backend/gl/src/command.rs

```diff
@@ -235,17 +244,94 @@ impl From<hal::Limits> for Limits {
     }
 }
 
-/// A command buffer abstraction for OpenGL.
-///
-/// If you want to display your rendered results to a framebuffer created externally, see the
-/// `display_fb` field.
 #[derive(Debug)]
-pub struct CommandBuffer {
+pub struct BufferStorage {
     pub(crate) memory: Arc<Mutex<BufferMemory>>,
     pub(crate) buf: BufferSlice,
     // Buffer id for the owning command pool.
     // Only relevant if individual resets are allowed.
     pub(crate) id: u64,
+}
+
+impl BufferStorage {
+    fn push_cmd(&mut self, cmd: Command) {
+        let mut memory = self
+            .memory
+            .try_lock()
+            .expect("Trying to record a command buffers, while memory is in-use.");
+
+        let cmd_buffer = match *memory {
+            BufferMemory::Linear(ref mut buffer) => &mut buffer.commands,
```

Comment:
if you just do `=> buffer` here and `=> storage.get_mut(&self.id).unwrap()` in the other arm, it will be less verbose